src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -473,13 +473,13 @@ static Set<Haplotype> filterPileupHaplotypes(final List<Haplotype> onlyNewHaplot
                                                  final Map<Kmer, MutableInt> kmerReadCounts,
                                                  final int numPileupHaplotypes,
                                                  final int kmerSize ) {
-
         // sort by score = # of kmers in haplotype that appear in any read -- the '-' sign sorts greatest to least
         // TODO this code might use some normalizing based on haplotype length in the future
         return onlyNewHaplotypes.stream()
-                .sorted(Comparator.comparingLong(hap -> -kmerizeSequence(hap.getBases(), kmerSize).stream()
+                .distinct() // TODO: The input haplotypes are not always unique.  Is this a horrible code smell?
```

Comment:
though with representation issues i'm not surprised if you could accidentally re-create an existing haplotype by injecting a new allele into a bunch of the assembly haplotypes...

Reply:
Yeah, it's bad that this method is given a `List` with non-unique elements, but it's totally natural that the process of generating haplotypes would be redundant.  I'm going to use a `Set` instead of a `List` to deal with the problem upstream.